index.js

```diff
@@ -58,6 +58,7 @@ class ShopifyToken {
    * @param {String} options.apiKey The API Key for the app
    * @param {String} [options.accessMode] The API access mode
    * @param {Number} [options.timeout] The request timeout
+   * @param {https.Agent|http.Agent} [options.agent] The agent for all http/https calls
```

Comment:
The endpoint is HTTPS only so I think this should be only `https.Agent`, no?

Reply:
Actually there can be different kind of tunnelling using agents. 
All possible combinations are:
1. Http-Over-Http
2. Http-Over-Https
3. Https-Over-Http
4. Https-Over-Https
In `getAccessToken` 's use case point 2 & point 4 are valid & useful. 
Hence added the support for `http.Agent` & `https.Agent`

Reply:
The proxy can be HTTP or HTTPS but the endpoint is always HTTPs so I think it's 3 & 4 in your list. For example see https://github.com/delvedor/hpagent/tree/v1.2.0#usage and https://github.com/delvedor/hpagent/blob/v1.2.0/index.js#L64.